### ISSUE TITLE
Question to meaning of step_type

### ISSUE THREAD

**Issue by datistiquo** [tensorflow/agents#534]
Hey,

I really wonder why nobody has problems with this and/or why there is no clear description of this in the documents of the code. There, it is clear what it is, but not how to use it to make "predictions" of the network. It seems not straightforward to get the q-values of a QNetwork so easily as inside the call() of the networks is this step_type.

I want to get the logits of the network, so I do:

`q_values, _ = q_network(X, step_type=?)`

So easily like with keras. But, I really don't know what to place as step_type above (marked as ?)?

Can someone explains what this has to be if I want to apply the network for test data?

The step_type something describe if a observation is the first, middle of end of a episode, but how do I know this for new data?
 This is really weird, or I do not understand this right.

**Comment by ormandi** [comment]
Thank you for your question.

Here you can find an example of how the `q_network` is used in a DQN agent: https://github.com/tensorflow/agents/blob/master/tf_agents/agents/dqn/dqn_agent.py#L524-L533 Basically, the high level expectation is that you have a `time_step` which contains both the observation (i.e. input of the observation denoted by `X` in your example), and the information whether this is the `first`, `middle` or `last` element of an episode encoded in the `step_type` of the `time_step`.

You can read more about `time_step` in the following tutorial: https://github.com/tensorflow/agents/blob/master/docs/tutorials/1_dqn_tutorial.ipynb
